crates/router/src/core/payouts.rs

```diff
@@ -476,14 +476,38 @@ pub async fn payouts_retrieve_core(
     key_store: domain::MerchantKeyStore,
     req: payouts::PayoutRetrieveRequest,
 ) -> RouterResponse<payouts::PayoutCreateResponse> {
-    let payout_data = make_payout_data(
+    let mut payout_data = make_payout_data(
         &state,
         &merchant_account,
         &key_store,
         &payouts::PayoutRequest::PayoutRetrieveRequest(req.to_owned()),
     )
     .await?;
 
+    let payout_attempt = payout_data.payout_attempt.to_owned();
+
+    // Form connector data
+    let connector_call_type = get_connector_choice(
```

Comment:
We should call the underlying connector only when `force_sync` is true

Reply:
Also, do we need to run this logic again for retrieve call? We already have the connector where the payout was created, could we directly use that connector for doing a sync?

Reply:
This is being called to get the ConnectorData, as for using the already present connector, `payout_attempt.connector` is being passed as a parameter to set the call connector type as predetermined. Thus, we remove all the other cases of the call connector type